isofit/radiative_transfer/kernel_flows.py

```diff
@@ -0,0 +1,448 @@
+#
+#  Copyright 2019 California Institute of Technology
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# ISOFIT: Imaging Spectrometer Optimal FITting
+# Author: Niklas Bohn, urs.n.bohn@jpl.nasa.gov
+#         Jouni Susiluoto, jouni.i.susiluoto@jpl.nasa.gov
+#
+
+import logging
+
+import h5py
+import numpy as np
+import yaml
+
+from isofit import ray
+from isofit.configs.sections.radiative_transfer_config import (
+    RadiativeTransferEngineConfig,
+)
+from isofit.core.common import combos, spectral_response_function
+from isofit.radiative_transfer.radiative_transfer_engine import RadiativeTransferEngine
+
+Logger = logging.getLogger(__file__)
+
+# This is a tempoary key mapping until we align the KF and Isofit key names
+# and get KF to store the key names in the jld2 file
+KEYMAPPING = {
+    1: {
+        "name": "AERFRAC_2",
+        "default_grid": [0.06, 0.2, 0.4, 0.6, 0.8, 1.0],
+        "default": 0.06,
+    },
+    2: {
+        "name": "H2OSTR",
+        "default_grid": [0.05, 0.75, 1.5, 2.25, 3.0, 3.75, 4.5],
+        "default": 0.05,
+    },
+    3: {"name": "surface_elevation_km", "default_grid": [0.0, 1.5, 3.0, 4.5, 6.0]},
+    4: {"name": "observer_altitude_km", "default_grid": [0.5, 1, 2, 5, 10, 20, 100]},
+    5: {"name": "solar_zenith", "default_grid": [0.0, 15.0, 30.0, 45.0, 60.0, 75.0]},
+    6: {"name": "observer_zenith", "default_grid": [0.0, 10.0, 20.0, 30.0, 40.0]},
+    7: {
+        "name": "relative_azimuth",
+        "default_grid": [0.0, 30.0, 60.0, 90.0, 120.0, 150.0, 180.0],
+    },
+    8: {"name": "solar_azimuth", "default_grid": [1, 91, 181, 271]},
+    9: {"name": "observer_azimuth", "default_grid": [1, 91, 181, 271]},
+}
```

Comment:
@pgbrodrick Is there a reason for displacing solar and observer azimuth by 1 degree? And do we actually need them when having the relative azimuth available?

Reply:
No good reason, I can modify the grid.  The default grids are actually entirely deprecated how I believe, so I might just remove them entirely.  Good catch.

I kept all variables that are in the emulator source code, just in case different arrangements are used...but you're right that they shouldn't be needed.